Client/Frontend/Browser/FindInPageBar.swift

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+
+protocol FindInPageBarDelegate: class {
+    func findInPage(findInPage: FindInPageBar, didTextChange text: String)
+    func findInPage(findInPage: FindInPageBar, didFindPreviousWithText text: String)
+    func findInPage(findInPage: FindInPageBar, didFindNextWithText text: String)
+    func findInPageDidPressClose(findInPage: FindInPageBar)
+}
+
+class FindInPageBar: UIView {
+    weak var delegate: FindInPageBarDelegate?
+    private let searchText = UITextField()
+    private let matchCountView = UILabel()
+    private let previousButton = UIButton()
+    private let nextButton = UIButton()
+
+    var currentResult = 0 {
+        didSet {
+            matchCountView.text = "\(currentResult)/\(totalResults)"
+        }
+    }
+
+    var totalResults = 0 {
+        didSet {
+            matchCountView.text = "\(currentResult)/\(totalResults)"
+            previousButton.enabled = totalResults > 1
+            nextButton.enabled = previousButton.enabled
+        }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        backgroundColor = UIColor.whiteColor()
+
+        searchText.addTarget(self, action: "didTextChange:", forControlEvents: UIControlEvents.EditingChanged)
```

Comment:
Nit: Not sure if we're doing it everywhere but for action selectors I've been prefixing them with SEL just to clarify that they are used as selectors like this.